R/eda.R

```diff
@@ -28,20 +28,73 @@ generate_report <- function(dataframe, cat_vars, num_vars) {
 #'
 #' @param dataframe tbl. The dataframe to be inspected.
 #' @param num_vars vector of character strings of the names of the numeric variables.
-#' @param plot logical. If TRUE (the default) the distribution of the numeric variables will be plotted.
 #'
-#' @return tbl. A dataframe contains the statistical summary.
+#' @return list. A list of a dataframe contains the statistical summary and the ggplot objects of the faceted histogram of numeric variables.
 #'
 #' @export
 #'
 #' @examples
-#' X <- tibble(type = c('Car','Bus','Car'), height = c(10,20,30))
-#' num_vars <- c('height')
-#' describe_num_var(X, num_vars)
-#'
-describe_num_var <- function(dataframe, num_vars, plot=TRUE) {
-  #TODO implement function
-  print(num_vars)
+#' X <- tibble(type = c('Car', 'Bus', 'Car'), height = c(10, 20, 15), width = c(10, 15, 13))
+#' num_vars <- c('height', 'width')
+#' result <- describe_num_var(X, num_vars)
+#' result$summary
+#' result$plot
+#'
+describe_num_var <- function(dataframe, num_vars) {
+  # Check the input dataframe is a dataframe
+  if (!is.data.frame(dataframe)) {
+    stop("The first argument should be a data frame.")
+  }
+
+  # Transform the input into tibble
+  dataframe <- as_tibble(dataframe)
+
+  # Chech the input num_vars is a vector of characters
+  if (!is.character(num_vars) | !is.vector(num_vars)) {
+    stop("The second argument should be a vector of characters.")
+  }
+
+  # Check the input num_vars is a vector of column names of dataframe
+  if (!all(num_vars %in% colnames(dataframe))) {
+    stop("The second argument should be a subset of the cloumn names of the dataframe.")
+  }
+
+  # Select the numeric variables to work with
+  df <- dataframe %>%
+    select(num_vars)
+
+  # Check if only the numeric variables are selected
+  if (!all(sapply(df, is.numeric))) {
+    stop("Not only the numeric variables are selected, please check the input.")
```

Comment:
```suggestion
    stop("Only numeric columns expected, please check the input.")
```
You can use this or change it to something more meaningful.